src/main/java/org/terasology/ligthandshadow/barrier/MagicDomeSystem.java

```diff
@@ -77,17 +78,40 @@ public void onCharacterMovement(CharacterMoveInputEvent moveInputEvent, EntityRe
         float distance = pos.length();
 
         float deltaDistance = TeraMath.fastAbs(pos.distance(lastPos));
-        if (deltaDistance > 0.2f) {
-            logger.info("CharacerMoveInputEvent: position: {} - distance from O: {}, delta: {}", pos, distance, deltaDistance);
-            lastPos.set(pos);
 
-            if (distance > WORLD_RADIUS) {
-                logger.info("Sending player back!");
-                Vector3f impulse = pos.normalize().invert();
-                impulse.set(impulse.scale(64).setY(6));
-                player.send(new CharacterImpulseEvent(impulse));
+        for (EntityRef domeEntity : entityManager.getEntitiesWith(MagicDome.class, LocationComponent.class)) {
+            LocationComponent domeLocationComponent = domeEntity.getComponent(LocationComponent.class);
+            Vector3f domeCenter = domeLocationComponent.getWorldPosition();
+            Vector3f domeVerticalTop = domeCenter.addY(WORLD_RADIUS);
 
-                player.send(new PlaySoundEvent(magicDomeEntity.getComponent(MagicDome.class).hitSound, 2f));
+            MagicDome dome = domeEntity.getComponent(MagicDome.class);
+
+            if (deltaDistance > 0.2f) {
```

Comment:
Here, we might check the player's velocity to catch a teleport event and don't act on it. Maybe you can give it a try?